pydm/widgets/timeplot.py

```diff
@@ -187,35 +445,75 @@ def set_needs_redraw(self):
 
     @Slot()
     def redrawPlot(self):
+        """
+        Redraw the graph. If the display UI is provided, also check if the current curve is receiving data.
+        """
         if not self._needs_redraw:
             return
+
         self.updateXAxis()
+
         for curve in self._curves:
             curve.redrawCurve()
+            if self.plot_display:
+                self.plot_display.update_curve_data(curve)
```

Comment:
This should probably be done with a signal too.  I am kind of worried about emitting a signal for every curve, on every redraw, though.  I'd suggest that instead of making any new signals, the Display class that owns the timeplot should just connect to the `TimePlotCurveItem.data_changed` signal for each curve on the plot.

Reply:
Now using a signal. Since the monitoring is at the plot level, I'm still emitting the signal at the redraw. The purpose here is to grey out the curve label if it doesn't have a valid value to draw (NaN). Checking at redraw lets the Display know if a PV is disconnected.